File: SistemaVendas/src/br/com/k21/CalculadoraRoyalties.java
package br.com.k21;

import java.util.List;

import br.com.k21.dao.VendaRepository;
import br.com.k21.modelo.Venda;

public class CalculadoraRoyalties {

	private CalculadoraComissao calculadoraComissao;
	private VendaRepository vendaRepository;
	
	public CalculadoraRoyalties(CalculadoraComissao calculadoraComissao,
			VendaRepository vendaRepository) {
		this.calculadoraComissao = calculadoraComissao;
		this.vendaRepository = vendaRepository;
	}

	public double calcularRoyalties(int mes, int ano) {
		List<Venda> vendas = this.vendaRepository.obterVendasPorMesEAno(ano, mes);
		
		Venda venda = vendas.get(0);
		
		double valorVenda = venda.getValor();
		
		return 0.2* (valorVenda-this.calculadoraComissao.calcular(valorVenda));
		
	}

}

File: Calculator-PHPUnit/src/CalculadoraRoyalties.php
<?php

require_once 'CalculadoraComissao.php';
require_once 'VendaRepository.php';
require_once 'Venda.php';

/**
 * @author cfc
 */
class CalculadoraRoyalties
{
    private $repository;
    
    function __construct(VendaRepository $repository) {
        $this->repository = $repository;
    }
    
    public function calcularRoyalties($mes, $ano)
    {
        
        $listaVendas = $this->repository->obterVendasPorMesEAno($ano, $mes);
        
        $totalBruto = 0;
        $totalComissao = 0;
        
        foreach ((array)$listaVendas as $venda) {
            $totalBruto += $venda->valor;
            $totalComissao += CalculadoraComissao::calcular($venda->valor);
        }
        
        return ($totalBruto - $totalComissao) * 0.2;
    }
}
?>